### PR TITLE
Remove unused `ember.min.js` build asset

### PR DESCRIPTION
**Pull request by Turbo87**
According to https://github.com/emberjs/ember.js/pull/17275#issuecomment-445059036 the `ember.min.js` asset is not being used anymore. This PR removes the asset from the build pipeline, since the uglify step takes a significant amount of time.

/cc @rwjblue

### CODE REVIEW

**Review by rwjblue — APPROVED** [review]

File: bin/run-tests.js [hunk 1]
```diff
@@ -90,9 +90,6 @@ function generateBuiltTests() {
   testFunctions.push(function() {
     return run(common + '&nolint=true');
   });
-  testFunctions.push(function() {
-    return run(common + '&dist=min&prod=true');
```
Comment (rwjblue) [inline bin/run-tests.js]: Hmm, this reminds me, once upon a time we had legit issues when minified so we started running our test suite against the minified assset.

I'm not 100% sure that is still a valid concern/issue, but it warrants further thought...

**Review by rwjblue — CHANGES_REQUESTED** [review]
Will add this for review with the rest of the team to confirm...

File: bin/run-tests.js (hunk 1 above)
Comment (Turbo87) [reply bin/run-tests.js]: sounds like a valid concern, but as you said Ember.js is no longer controlling its own minification process, so I'm not sure it's helpful to run the tests against just that single version of `terser` while out in the wild other versions and possibly even other minifiers are being used 🤔

Comment (rwjblue) [reply bin/run-tests.js]: Yep, totally agree.

**Comment by tomdale** [comment]
As discussed in the core team meeting today, we are in favor of removing this but want to make sure we have _some_ sort of end-to-end smoke test of Ember and Ember CLI working together, to catch minifier bugs but also the many other categories of regressions that can happen at integration points.

Until we get something like that in place, would it be possible to modify the emitted minified version to produce a `console.warn` that the minified version is deprecated and will stop being produced in the future? Even though I don't think anyone is actually using this file, if there are folks they'll at least get a bit of heads up.

**Comment by Turbo87** [comment]
> Until we get something like that in place, would it be possible to modify the emitted minified version to produce a `console.warn` that the minified version is deprecated and will stop being produced in the future?

should be possible

I'll close this PR for now as we'll probably keep the minified version around for a few releases before we can remove it.

**Pull request closed without merging**